python/cudf/cudf/_lib/pylibcudf/column_factories.pyx

```diff
@@ -0,0 +1,266 @@
+                )
+    elif MaskArg is tuple:
+        mask_buf, null_count = mstate
+        mask = move(mask_buf.c_release())
+
+        with nogil:
+            result = move(
+                cpp_make_fixed_point_column(
+                    type_.c_obj,
+                    size,
+                    move(mask),
+                    null_count
+                )
+            )
+    else:
+        raise TypeError("Invalid mask argument")
+
+    return Column.from_libcudf(move(result))
+
+cpdef Column make_timestamp_column(
+    DataType type_,
+    size_type size,
+    MaskArg mstate
+):
+
+    cdef unique_ptr[column] result
+    cdef mask_state state
+    cdef DeviceBuffer mask_buf
+    cdef device_buffer mask
+    cdef size_type null_count
+
+    if MaskArg is object:
+        if isinstance(mstate, MaskState):
+            state = mstate
+            with nogil:
+                result = move(
+                    cpp_make_timestamp_column(
+                        type_.c_obj,
+                        size,
+                        state
+                    )
+                )
+    elif MaskArg is tuple:
+        mask_buf, null_count = mstate
+        mask = move(mask_buf.c_release())
+
+        with nogil:
+            result = move(
+                cpp_make_timestamp_column(
+                    type_.c_obj,
+                    size,
+                    move(mask),
+                    null_count
+                )
+            )
+    else:
+        raise TypeError("Invalid mask argument")
+
+    return Column.from_libcudf(move(result))
+
+cpdef Column make_duration_column(
+    DataType type_,
+    size_type size,
+    MaskArg mstate
+):
+
+    cdef unique_ptr[column] result
+    cdef mask_state state
+    cdef DeviceBuffer mask_buf
+    cdef device_buffer mask
+    cdef size_type null_count
+
+    if MaskArg is object:
+        if isinstance(mstate, MaskState):
+            state = mstate
+            with nogil:
+                result = move(
+                    cpp_make_duration_column(
+                        type_.c_obj,
+                        size,
+                        state
+                    )
+                )
+    elif MaskArg is tuple:
+        mask_buf, null_count = mstate
+        mask = move(mask_buf.c_release())
+
+        with nogil:
+            result = move(
+                cpp_make_duration_column(
+                    type_.c_obj,
+                    size,
+                    move(mask),
+                    null_count
+                )
+            )
+    else:
+        raise TypeError("Invalid mask argument")
+
+    return Column.from_libcudf(move(result))
+
+cpdef Column make_fixed_width_column(
+    DataType type_,
+    size_type size,
+    MaskArg mstate
+):
+
+    cdef unique_ptr[column] result
+    cdef mask_state state
+    cdef DeviceBuffer mask_buf
+    cdef device_buffer mask
+    cdef size_type null_count
+
+    if MaskArg is object:
+        if isinstance(mstate, MaskState):
```

Comment:
if this `if` statement branch is not taken, is it supposed to raise `TypeError`?
(similar comment for other places as well).

Reply:
Thanks - caught a segfault here 👍